tests/test_io.py

```diff
@@ -23,28 +26,115 @@ def captured_output():
+    def test_log_text_with_key_in_global_PROGTEXT(self, setup_data):
+        """
+        Testing standard output of a single text without expansion.
+        """
+        key = "confirm"
+        with mock.patch("builtins.open", mock.mock_open(read_data="[confirm]\r\nConfirm key found\r\n")) as mock_file:
+            dfastmi.io.load_program_texts("")
+        with captured_output() as (out, err):
+            dfastmi.io.log_text(key)
+        outstr = out.getvalue().splitlines()
+        strref = "Confirm key found"
+        assert outstr[0] == strref
 
-@pytest.fixture
-def setup_data():
-    dfastmi.io.load_program_texts("dfastmi/messages.UK.ini")
+    def test_log_text_with_key_and_variable_id_in_global_PROGTEXT(self, setup_data):
+        """
+        Testing standard output of a single text without expansion.
+        """
+        key = "confirm"
+        dict = {"value": "ABC"}
+        with mock.patch("builtins.open", mock.mock_open(read_data="[confirm]\r\nConfirm key found with {value}\r\n")) as mock_file:
+            dfastmi.io.load_program_texts("")
+        with captured_output() as (out, err):
+            dfastmi.io.log_text(key,dict=dict)
+        outstr = out.getvalue().splitlines()
+        strref = "Confirm key found with ABC"
+        assert outstr[0] == strref
+    
+    def test_log_text_two_times_with_key_and_variable_id_in_global_PROGTEXT(self, setup_data):
+        """
+        Testing standard output of a single text without expansion.
+        """
+        key = "confirm"
+        dict = {"value": "ABC"}
+        with mock.patch("builtins.open", mock.mock_open(read_data="[confirm]\r\nConfirm key found with {value}\r\n")) as mock_file:
+            dfastmi.io.load_program_texts("")
+        with captured_output() as (out, err):
+            dfastmi.io.log_text(key, dict=dict, repeat=2)
+        outstr = out.getvalue().splitlines()
+        strref = "Confirm key found with ABC"
+        assert outstr[0] == strref
+        assert outstr[1] == strref
 
-class Test_log_text():
-    def test_log_text_01(self, setup_data):
+    def test_log_text_with_key_and_variable_id_in_global_PROGTEXT_write_in_file(self, setup_data):
+        """
+        Testing standard output of a single text without expansion.
+        """
+        key = "confirm"
+        dict = {"value": "ABC"}
+        with mock.patch("builtins.open", mock.mock_open(read_data="[confirm]\r\nConfirm key found with {value}\r\n")) as mock_file:
+            dfastmi.io.load_program_texts("")
+
+        with mock.patch("builtins.open") as mock_file:
+            dfastmi.io.log_text(key, dict=dict, file=mock_file)
+            assert mock_file.write.called
+            mock_file.write.assert_called_once_with("Confirm key found with ABC\n")    
+
+class Test_data_access_log_text():
+    @pytest.fixture
+    def setup_data(self):
+        dfastmi.io.load_program_texts("dfastmi/messages.UK.ini")
+
+    def test_log_text_check_content_messages_uk(self, setup_data: None):
```

Comment:
why is the typehint none?

Reply:
typehint is removed in As discussed with @MRVermeulenDeltares, this should be resolved in [DFAST-90](https://issuetracker.deltares.nl/browse/DFAST-90)